core/src/SYCL/Kokkos_SYCL_Parallel_Range.hpp

```diff
@@ -111,12 +116,15 @@ class Kokkos::Impl::ParallelFor<FunctorType, Kokkos::RangePolicy<Traits...>,
     Kokkos::Experimental::Impl::SYCLInternal::IndirectKernelMem&
         indirectKernelMem = m_policy.space()
                                 .impl_internal_space_instance()
-                                ->m_indirectKernelMem;
+                                ->get_indirect_kernel_mem();
 
     const auto functor_wrapper = Experimental::Impl::make_sycl_function_wrapper(
         m_functor, indirectKernelMem);
+    // TODO joe: m_copy_event isn't relevant for a trivially copyable kernel
+    // does this matter?
     sycl::event event =
-        sycl_direct_launch(m_policy, functor_wrapper.get_functor());
+        sycl_direct_launch(m_policy, functor_wrapper.get_functor(),
+                           indirectKernelMem.m_copy_event);
```

Comment:
I would expect that we need to wait for `indirectKernelMem.m_copy_event` to be finished before leaving this function (at least with the current implementation of `USMObjectMem`.

Reply:
The staging area avoids this necessity.